railties/lib/rails/generators/app_base.rb

```diff
@@ -140,6 +141,70 @@ def build(meth, *args) # :doc:
         builder.public_send(meth, *args) if builder.respond_to?(meth)
       end
 
+      def deduce_implied_options(options, option_reasons, meta_options)
+        active = options.transform_values { |value| [] if value }.compact
+        irrevocable = (active.keys - meta_options).to_set
+
+        deduction_order = TSort.tsort(
+          ->(&block) { option_reasons.each_key(&block) },
+          ->(key, &block) { option_reasons[key]&.each(&block) }
+        )
+
+        deduction_order.each do |name|
+          reasons = option_reasons[name]&.select(&active).presence
+          active[name] ||= reasons if reasons
+          irrevocable << name if reasons&.any?(irrevocable)
+        end
+
+        revoked = options.select { |name, value| value == false }.keys.to_set - irrevocable
+        deduction_order.reverse_each do |name|
+          revoked += option_reasons[name].to_a if revoked.include?(name)
+        end
+        revoked -= meta_options
+
+        active.filter_map do |name, reasons|
+          reasons -= revoked.to_a
+          [name, reasons] unless revoked.include?(name) || reasons.empty?
+        end.to_h
+      end
+
+      OPTION_IMPLICATIONS = { # :nodoc:
+        (:skip_active_job)     => [:skip_action_mailer, :skip_active_storage],
+        (:skip_active_record)  => [:skip_active_storage],
+        (:skip_active_storage) => [:skip_action_mailbox, :skip_action_text],
+        (:skip_javascript)     => [:skip_hotwire],
+      }
+
+      def imply_options(option_implications = OPTION_IMPLICATIONS, meta_options: [])
+        option_reasons = {}
```

Comment:
Should this hash have empty hashes as default?

Reply:
`option_reasons` is a hash of arrays (`Hash<String, Array<String>>`), so the default would be empty arrays.  But, even though empty arrays would work, I think a default value makes things less clear when iterating over keys (e.g. [`option_reasons.each_key(&block)`](https://github.com/rails/rails/pull/45520/files#diff-a83245368df35c610a37260d0a51041de89211857d47cfd0b2d1301a11709164R149)).  But I don't have a strong opinion either way.

Reply:
It is fine this way.